### PR TITLE
Rubocop rules extension 2

### PR DESCRIPTION
**Pull request by danielkryska**

### CODE REVIEW

File: app/controllers/concerns/service/recommendable.rb [hunk 1]
```diff
@@ -87,8 +87,6 @@ def get_service_search_state
 
   def get_services_size_by(ab_test_version)
     case ab_test_version
-    when "v1"
-      3
```
Comment (danielkryska) [inline app/controllers/concerns/service/recommendable.rb:91]: By default services size is 3

**Review by michal-szostak — APPROVED** [review]

File: app/controllers/concerns/service/recommendable.rb (hunk 1 above)
Comment (jswk) [reply app/controllers/concerns/service/recommendable.rb:91]: I guess OK, although having explicit values for test versions is a nice thing.
I would consider going for the following, which neither triggers Rubocop nor removes the explicit values for test versions:
```ruby
  def get_services_size_by(ab_test_version)
    { "v1": 3, "v2": 2 }[ab_test_version] || 3
  end
```

File: app/views/providers/details/_object.html.haml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-- Array(provider.send(clazz)).map.with_index do |object, idx|
+- Array(provider.send(clazz)).map.with_index do |object, _idx|
```
Comment (jswk) [inline app/views/providers/details/_object.html.haml:1]:
```ruby
Array(provider.send(clazz)).map do |object|
```

File: app/views/providers/sidebar/_object.html.haml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 %ul.list-group
-  - Array(provider.send(clazz)).map.with_index do |object, idx|
+  - Array(provider.send(clazz)).map.with_index do |object, _idx|
```
Comment (jswk) [inline app/views/providers/sidebar/_object.html.haml:2]:
```ruby
Array(provider.send(clazz)).map do |object|
```

File: app/views/services/details/_object.html.haml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-- Array(service.send(clazz)).map.with_index do |object, idx|
+- Array(service.send(clazz)).map.with_index do |object, _idx|
```
Comment (jswk) [inline app/views/services/details/_object.html.haml:1]:
```ruby
Array(service.send(clazz)).map do |object|
```

File: app/views/services/_active_filters.html.haml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 .col-md-12.mb-2
   %h1.categories
-    - unless params[:q].blank?
+    - if !params[:q].blank?
```
Comment (jswk) [inline app/views/services/_active_filters.html.haml:3]:
```ruby
if params[:q].present?
```

File: config/initializers/rswag-ui.rb [hunk 6]
```diff
@@ -1,5 +1,3 @@
-# frozen_string_literal: true
-
```
Comment (jswk) [inline config/initializers/rswag-ui.rb:2]: Why remove this?

File: spec/requests/api/v1/omses/messages_controller_spec.rb [hunk 7]
```diff
@@ -321,7 +321,11 @@
         schema "$ref" => "error.json"
         let(:oms_id) { 1 }
         let(:project_id) { 1 }
+
+        # rubocop:disable Lint/EmptyBlock
         let(:message_payload) {}
```
Comment (jswk) [inline spec/requests/api/v1/omses/messages_controller_spec.rb:326]: Couldn't these be just removed?

**Review by Marcelinna — CHANGES_REQUESTED** [review]
PR errored resource
image
Beta
image

File: spec/requests/api/v1/omses/messages_controller_spec.rb (hunk 7 above)
Comment (danielkryska) [reply spec/requests/api/v1/omses/messages_controller_spec.rb:326]: missing param cause error:

```
NoMethodError: undefined method `message_payload' for #<RSpec::ExampleGroups::ApiV1OMSesMessagesController::ApiV1OmsOmsIdMessages::Post::UserNotRecognized "returns a 401 response" (/home/eland/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/rswag-specs-2.4.0/lib/rswag/specs/example_group_helpers.rb:97)>

  0) Api::V1::OMSes::MessagesController /api/v1/oms/{oms_id}/messages post user not recognized returns a 401 response
     Failure/Error: super

     NoMethodError:
       undefined method `message_payload' for #<RSpec::ExampleGroups::ApiV1OMSesMessagesController::ApiV1OmsOmsIdMessages::Post::UserNotRecognized "returns a 401 response" (/home/eland/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/rswag-specs-2.4.0/lib/rswag/specs/example_group_helpers.rb:97)>
     # <internal:kernel>:90:in `tap'
     # -e:1:in `<main>'

Comment (jswk) [reply spec/requests/api/v1/omses/messages_controller_spec.rb:326]: Aah, ok. I assumed it's being defined somewhere above.
So maybe:
```ruby
        let(:message_payload) { nil }
```
The block isn't empty then.

File: app/views/services/_errors.html.haml [hunk 8]
```diff
@@ -4,4 +4,5 @@
     .container
       - if service&.sources&.first&.errored.present?
         .alert.alert-danger.mb-0.text-center
-          = ("This resource is not up to date because of invalid import data. Reason: #{service.sources.first.errored}")
+          = _("This resource is not up to date because of invalid import data. ",
+              "Reason: #{service.sources.first.errored}")
```
Comment (Marcelinna) [inline app/views/services/_errors.html.haml:8]: .

File: app/controllers/concerns/service/recommendable.rb (hunk 1 above)
Comment (danielkryska) [reply app/controllers/concerns/service/recommendable.rb:91]: done :)

File: app/views/services/_active_filters.html.haml [hunk 9]
```diff
@@ -1,6 +1,6 @@
 .col-md-12.mb-2
   %h1.categories
-    - unless params[:q].blank?
+    - if params[:q]&.present?
```
Comment (jswk) [inline app/views/services/_active_filters.html.haml:3]: Ok, but the safe navigation `&.` is unnecessary.
```
pry(main)> {}[:c].present?
=> false
```

Comment (danielkryska) [reply app/views/services/_active_filters.html.haml]: done

**Review by jswk — APPROVED** [review]

File: spec/requests/api/v1/omses/messages_controller_spec.rb (hunk 7 above)
Comment (danielkryska) [reply spec/requests/api/v1/omses/messages_controller_spec.rb]: done

File: app/views/services/configurations/attributes/_multiselect.html.haml [hunk 10]
```diff
@@ -3,7 +3,6 @@
   - has_errors = attr.errors[:id].present?
   - valid = !has_errors && attr.value
   = field.input_field attr.id,
-    id: "parameter_#{attr.id}",
     class: "form-check-input #{"is-invalid" if has_errors} #{"is-valid" if valid}",
     collection: attr.config["values"].map { |e| ["#{e} #{attr.unit}", e] },
```
Comment (Marcelinna) [inline app/views/services/configurations/attributes/_multiselect.html.haml:7]: id: "parameter_#{attr.id}", it is not necessary?

File: config/environments/production.rb [hunk 11]
```diff
@@ -130,7 +130,7 @@
 
   raise "Missing ASSET_HOST" if ENV["ASSET_HOST"].blank?
   raise "Missing ASSET_PROTOCOL" if ENV["ASSET_PROTOCOL"].blank?
-  config.action_controller.asset_host = "#{ENV["ASSET_HOST"]}"
+  config.action_controller.asset_host = (ENV["ASSET_HOST"]).to_s
   config.action_mailer.asset_host = "#{ENV["ASSET_PROTOCOL"]}://#{ENV["ASSET_HOST"]}"
 
   # SMTP settings
```
Comment (Marcelinna) [inline config/environments/production.rb:136]: it is ok?

File: app/views/services/configurations/attributes/_multiselect.html.haml (hunk 10 above)
Comment (jswk) [reply app/views/services/configurations/attributes/_multiselect.html.haml:7]: I'll have a look, although it seems that apart from a few places setting the `id` on configuration attributes adds more problems than profit (checkbox and select labels don't work as expected then).

File: config/environments/production.rb (hunk 11 above)
Comment (jswk) [reply config/environments/production.rb:136]: Yeah, it's equivalent, though not enforced by linter formatter anyway. I will scrap this one in the next push.

File: app/views/services/_errors.html.haml (hunk 8 above)
Comment (jswk) [reply app/views/services/_errors.html.haml]: It should work now, please resolve if handled properly now.

File: app/views/services/configurations/attributes/_multiselect.html.haml (hunk 10 above)
Comment (jswk) [reply app/views/services/configurations/attributes/_multiselect.html.haml:7]: Ok, `id`s in most of the attributes weren't necessary. I've stripped them altogether. Have a look.

**Comment by Marcelinna** [comment]
When I fill field Hosting legal entity in Provider creation form
pr
image
image
beta
image

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**